index.html

```diff
@@ -37,304 +37,179 @@
+        Examples of usage
       </h2>
       <p>
-        Example 1 demonstrates decompressing a Blob to ImageData via an
-        intermediate canvas. Note the steps involving the canvas, notably
-        <code>getImageData</code>, are synchronous.
-      </p>
-      <pre class='example highlight'>
-// note: looks asynchronous because it necessarily returns a promise
-// due to loading the blob as an Image, but the important parts are
-// still synchronous.
-function BlobToImageData(blob)
-{
-        let blobUrl = URL.createObjectURL(blob);
-
-        return new Promise((resolve, reject) =&gt;
-        {
-                let img = new Image();
-                img.onload = () =&gt; resolve(img);
-                img.onerror = err =&gt; reject(err);
-                img.src = blobUrl;
-        }).then(img =&gt;
-        {
-                URL.revokeObjectURL(blobUrl);
-
-                let w = img.width;
-                let h = img.height;
-
-                let canvas = document.createElement("canvas");
-                canvas.width = w;
-                canvas.height = h;
-                let ctx = canvas.getContext("2d");
-                ctx.drawImage(img, 0, 0);
-                return ctx.getImageData(0, 0, w, h);    // some browsers synchronously decode image here
-        });
-};
-</pre>
-      <p>
-        With this proposal it can be simplified to:
-      </p>
-      <pre class='example highlight'>
-function BlobToImageData(blob)
-{
-        return ImageData.create(blob);
-};
-</pre>
-      <p>
-        Example 3 demonstrates compressing an ImageData to a Blob via an
-        intermediate canvas.
+        This demonstrates decompressing a <code>Blob</code> to
+        <code>ImageData</code>.
       </p>
       <pre class='example highlight'>
-function ImageDataToBlob(imageData)
-{
-        let w = imageData.width;
-        let h = imageData.height;
-        let canvas = document.createElement("canvas");
-        canvas.width = w;
-        canvas.height = h;
-        let ctx = canvas.getContext("2d");
-        ctx.putImageData(imageData, 0, 0, w, h);        // synchronous
-
-        return new Promise((resolve, reject) =&gt;
-        {
-                canvas.toBlob(resolve); // implied image/png format
-        });
-};
+ImageData.create(blob).then(
+  imgData =&gt; doSomethingWith(imgData)
+);
 </pre>
       <p>
-        With the new method this can be simplified to:
+        This example demonstrates compressing an <code>ImageData</code> to a
+        <code>Blob</code>, and then displaying it in a
+        <code>HTMLImageElement</code>.
       </p>
       <pre class='example highlight'>
-function ImageDataToBlob(imageData)
-{
-        return imageData.toBlob();              // implied "image/png" format
-};
+var type = (Image.canEncodeType("image/jpg")) ? "image/jpg" : "image/png";
+imageData.toBlob(type).then((blob) =&gt; {
+  const img = new Image();
+  img.src = URL.createObjectURL(blob);
+  document.body.appendChild(img);
```

Comment:
Appending children to the end of body from async funcs can be... not so great, since you're racing against other scripts adding things to the DOM.